configure

```diff
@@ -3159,47 +3159,65 @@ EOF
 		append_var LIBS "-lnds9"
 		;;
 	emscripten)
-		# mandatory emscripten flags
-		append_var LDFLAGS "-s ALLOW_MEMORY_GROWTH=1 -s ASYNCIFY -s FORCE_FILESYSTEM=1"
-
 		append_var DEFINES "-DEMSCRIPTEN"
 		add_line_to_config_mk 'EMSCRIPTEN = 1'
+		append_var LDFLAGS "-s FORCE_FILESYSTEM"
+		append_var LDFLAGS "-s INITIAL_MEMORY=128MB -s TOTAL_STACK=32MB -s ALLOW_MEMORY_GROWTH=1"
+		append_var LDFLAGS "-s ASYNCIFY=1 -s ASYNCIFY_STACK_SIZE=1048576"
 
 		if test "$_debug_build" = yes; then
-			_optimization_level=-O2
+			_optimization_level=-O2 
 			append_var LDFLAGS "-O2 -g3 -s ASSERTIONS=2"
 		else
 			_optimization_level=-O3
 			append_var LDFLAGS "-O3"
 		fi
 
 		# activate emscripten-ports
-		if test "$_freetype2" != no; then
-			append_var LDFLAGS "-s USE_FREETYPE=1 -s SUPPORT_LONGJMP=1" # freetype requires setjmp
+		if test "$_sdl" != no; then # we enable SDL2 by default 
+			append_var LDFLAGS "-s USE_SDL=2 "
+		fi
+
+		# We explicitly disable optional libraries if not enabled. "auto" would depend
+		# on whether the port has been used before (and is detected) which is unpredictable.
+		if test "$_freetype2" = yes; then
 			# neither pkg-config nor freetype-config work, so we setup freetype manually
-			_freetypepath="$EMSCRIPTEN/cache/ports-builds/freetype/"
-			FREETYPE2_CFLAGS="-I$_freetypepath/include" # there were link errors / missing symbols without this
+			append_var LDFLAGS "-s USE_FREETYPE=1"
 			_freetype_found="true"
+			_freetypepath="$EMSDK/upstream/emscripten/cache/sysroot/include/freetype2/"
+			FREETYPE2_CFLAGS="-I$_freetypepath"
 		else
-			#use link time optimization to further reduce exe size (this can't be used with setjmp which freetype requires)
-			# TODO: Figure out why this is a conflict and/or if freetype can be built without setjmp
-			append_var CXXFLAGS "-flto"
-			append_var LDFLAGS "-flto"
+			_freetype2=no 
 		fi
-		if test "$_jpeg" != no; then
+		if test "$_gif" = yes; then
+		 	append_var LDFLAGS "-s USE_GIFLIB=1"
+		else
+			_gif=no 
+		fi
+		if test "$_jpeg" = yes; then
 			append_var LDFLAGS "-s USE_LIBJPEG=1"
+		else
+			_jpeg=no 
 		fi
-		if test "$_png" != no; then
-			append_var LDFLAGS "-s USE_LIBPNG=1"
+		if test "$_ogg" = yes; then
+			append_var LDFLAGS "-s USE_OGG=1" 
+		else
+			_ogg=no 
 		fi
-		if test "$_sdl" != no; then
-			append_var LDFLAGS "-s USE_SDL=2 "
+		if test "$_png" = yes; then 
+			append_var LDFLAGS "-s USE_LIBPNG=1"
+		else
+			_png=no 
 		fi
-		if test "$_vorbis" != no; then
-			append_var LDFLAGS "-s USE_OGG=1" # vorbis needs to be linked against OGG (even if we use an external vorbis lib)
+		if test "$_vorbis" = yes; then
+			append_var LDFLAGS "-s USE_VORBIS=1" 
+		else
+			_vorbis=no 
 		fi
-		if test "$_zlib" != no; then
+		if test "$_zlib" = yes; then 
```

Comment:
trailing whitespace

Reply:
fixed, thank you! (there was another one also in the check for `$_png` which I also fixed)